src/trainer_see.c

```diff
@@ -369,14 +369,26 @@ bool8 CheckForTrainersWantingBattle(void)
 
     for (i = 0; i < OBJECT_EVENTS_COUNT; i++)
     {
+        for (u8 j = 0; j < OBJECT_EVENTS_COUNT; j++)
+        {
+            k = 0;
+            if (gObjectEvents[j].localId == i)
+            {
+                k = j;
+                break;
+            }
+            else
+            {
+                continue;
+            }
+        }       
         u8 numTrainers;
```

Comment:
I think you need an `if (j == OBJECT_EVENTS_COUNT) continue;` or something to handle the case where this local ID isn't found so that we move onto searching the next local ID (atm we'll just take `gObjectEvents[0]`).